core/consensus/common/src/import_queue/basic_queue.rs

```diff
@@ -65,11 +62,27 @@ impl<B: BlockT> BasicQueue<B> {
 			finality_proof_import,
 		);
 
+		let mut pool = futures::executor::ThreadPool::builder()
+			.name_prefix("import-queue-worker-")
+			.pool_size(1)
+			.create()
+			.ok();
+
+		let manual_poll;
+		if let Some(pool) = &mut pool {
+			// TODO: this expect() can be removed once
+			// https://github.com/rust-lang-nursery/futures-rs/pull/1750 is merged and deployed
+			pool.spawn(future).expect("ThreadPool can never fail to spawn tasks; QED");
```

Comment:
Note that it is unfortunately no longer possible to properly recover from task spawning errors.
I went for creating a local `ThreadPool` here so that we are guaranteed to never have any error (as in: if creating the pool succeeds, then spawning the task will also succeed).
In the future, we should probably pass some task spawning utility by parameter, but the idiomatic way to do that remains to be discovered.